src/Visualization/Python/Render1D.py

```diff
@@ -0,0 +1,277 @@
+        data.append([pair[1] for pair in time_data[id_and_time]])
+
+    for h5file in h5files:
+        h5file.close()
+    return time, coords, data
+
+
+def render_single_time(args, coords, data):
+    '''
+    Renders a single time snap shot
+    :param args: the command line arguments
+    :return: None
+    '''
+    plt.xlabel("x")
+    plt.ylabel(args['var'])
+    plt.title("t = %.5f" % int(args['time']))
+    plt.plot(coords[int(args['time'])], data[int(args['time'])], 'o')
+    if args['save']:
+        print("Writing still to file %s.pdf" % args['save'])
+        plt.savefig("%s.pdf" % args['save'],
+                    transparent=True,
+                    format='pdf',
+                    bbox_inches='tight')
+    else:
+        plt.show()
+    return None
+
+
+def render_animation(args, time, coords, data):
+    '''
+    Render an animation of the data
+    :param args: the command line arguments
+    :return: None
+    '''
+    def init():
+        '''
+        Initialize the animation canvas
+        :return: empty line and title info
+        '''
+        line.set_data([], [])
+        title.set_text("")
+        return line, title
+
+    def animate(iteration):
+        '''
+        Update the animation canvas
+        '''
+        title.set_text("t = %.5f" % time[iteration])
+        line.set_data(coords[iteration], data[iteration])
+        return line, title
+
+    fig = plt.figure()
+    ax = plt.axes(xlim=(find_extrema_over_data_set(coords)),
+                  ylim=(find_extrema_over_data_set(data)))
+
+    line, = ax.plot([], [], 'o', lw=2)
+    ax.set_xlabel('x')
+    ax.set_ylabel(args['var'])
+    title = ax.set_title("")
+    my_interval = float(args['interval'])
+    anim = animation.FuncAnimation(fig,
+                                   animate,
+                                   init_func=init,
+                                   frames=len(time),
+                                   interval=my_interval)
+    if args['output']:
+        myfps = args['fps']
+        print("Writing animation to file %s.mp4 at %d frames per second" %
+              (args['output'], myfps))
+        anim.save(args['output'] + ".mp4", writer='ffmpeg')
+    else:
+
+        def propagate_exceptions(self, exc, val, tb):
+            raise exc(val).with_traceback(tb)
+
+        # Provides an error message with traceback when an exception is raised
+        tkinter.Tk.report_callback_exception = propagate_exceptions
+        plt.show()
+    return None
+
+
+def main(args):
+    '''
+    :param args: command line arguments
+    :return: None
+    '''
+
+    if args['list_vars']:
+        print_var_names(args)
+        sys.exit(0)
+
+    time, coords, data = get_data(args)
+    if args['time']:
+        render_single_time(args, coords, data)
+    else:
+        render_animation(args, time, coords, data)
+    return None
+
+
+if __name__ == "__main__":
+    main(parse_cmd_line())
```

Comment:
You don't need the extra function `main`. You can move the body of that function here.